### DISCUSSION TITLE
Feature Suggestion: Gizmo visibility changing when going through objects

### DISCUSSION THREAD

**Discussion by Vasquarde** [godotengine/godot-proposals#9466] [Editor]
I'd like to see more visual indication of when the gizmo is going through objects such as colliders and meshes. Maybe something like a color change or some other obvious sign the gizmo is clipping through something. Same for when rotating and scaling. This would allow for more on the fly precise placement of things.

**Comment by Calinou** [comment]
This is already done on the orange 3D selection box by drawing it twice:

- Once with 50% opacity and depth test disabled.
- Once with 100% depth opacity with standard depth test.

We could do this for more 3D gizmos, but it requires a custom implementation in each of those. Also, doubling draw calls and vertex count can have a performance impact in large scenes, especially on low-end hardware.

I think this makes sense to do for the 3D editor manipulation gizmo (translate/rotate/scale), but I wouldn't do it for *every* 3D gizmo due to the aforementioned performance reasons.

**Comment by Vasquarde** [comment]
Yeah I meant for the 3D editor manipulation gizmo, sorry if I didn't clarify. 

There's no clear distinction for the gizmo going through an editor object. My suggestion would simply to have some way similar to how colliders and other editor icons can visibly go through editor objects.
